src/external/dftk_to_wannier90.jl

```diff
@@ -0,0 +1,605 @@
+using ProgressMeter
+using LinearAlgebra
+using Dates
+
+## Works in the following order
+# $ dftk2wan_win_file(...) to generate the win file
+# $ wannier90.x -pp "prefix" ------------ nnkp file
+# $ dftk2wan_wannierization_files(...)--- mmn,amn,eig files
+# $ wannier90.x "prefix"  --------------- actual wannierization
+
+## Remark : the routines "dict_AOs", "guess_win", "cart_to_spherical" and "A_k_matrix_win_guesses" and the "projs" table in "read_nnkp_file"
+# are dedicated to computation of A_k matrices with Wannier90 guess, read from the nnkp file.
+# It is bugged and may not be usefull but I left it there.
+
+## TODO
+# 1 - check if kgrid can be removed from dftk2wan_win_file arguments
+# 2 - add handling of the SCDM routine as an alternative guess
+# 3 - optimize the "generate_mmn" routine
+
+
+"""
+    Create a .win file for Wannier90, compatible with the system studied with DFTK.
+    Options have to be compeleted by hand.
+"""
+function dftk2wan_win_file(prefix::String,basis::PlaneWaveBasis,scfres,num_wann;
+                           kgrid = [4,4,4], #can we retrieve this from scfres or basis ? Feel this should not be an argument.
+                           bands_plot = false,
+                           num_iter = 500,
+                           num_print_cycles = 50
+                           )
+    
+    ## Check kgrids
+    if (prod(kgrid) != size(scfres.ψ,1))
+        error("The given kgrid doesn't match the one used for the scf calculation")
+    end    
+
+    ## Write in file   
+    open("$prefix.win","w") do f
+
+        write(f,"! $prefix.win file generated by DFTK at "*string(now())*"\n"^2)
+        
+        ## General parameters
+        num_bands = size(scfres.ψ[1],2)
+        write(f,"num_bands "*(" "^9)*"= $num_bands"*"\n")
+        write(f,"num_wann "*(" "^10)*"= $num_wann"*"\n"^2)
+
+        write(f,"num_iter "*(" "^10)*"= $num_iter"*"\n")
+        write(f,"num_print_cycles "*(" "^2)*"= $num_print_cycles"*"\n"^2)
+        
+        ## Messages for the user
+        write(f,"!"^20*" Complete with optional parameters :"*"\n"^2)
+
+
+
+        ## System
+        write(f,"!"^20*" System"*"\n"^2)
+
+        # Unit cell block
+        write(f,"begin unit_cell_cart"*"\n"*"bohr"*"\n")
+        unit_cell = transpose(basis.model.lattice)         #unit cell vectors are in raws in Wannier90.
+        for i in 1:3
+            for j in 1:3
+                write(f,"$(unit_cell[i,j]) ")
+            end
+            write(f,"\n")
+        end
```

Comment:
can't you just write unit_cell here?

Reply:
I tried but the output is not a string and string(unit_cell) writes "[line 1; line 2; line 3]".
I managed to change that with the @sprintf macro however. I think it is a bit better but there is still one "for" loop over the lines of the unit_cell.

line_sprintf(l) = @sprintf("%.6f %.6f %.6f \n",l[1],l[2],l[3])
for i in 1:3
      write(f,line_sprintf(unit_cell[i,:]))
end

Reply:
I would just do:
```
for uc in eachcol(unit_cell)
    write(f, uc..., "\n")
end
```

Reply:
Indeed works fine !